db/comdb2.c

```diff
@@ -1899,6 +1925,66 @@ int init_check_constraints(dbtable *tbl)
     return 0;
 }
 
+static int resize_reverse_constraints(struct dbtable *db, size_t newsize)
+{
+    constraint_t **temp = realloc(db->rev_constraints, newsize * sizeof(constraint_t *));
+    if (temp) {
+        db->rev_constraints = temp;
+        db->cap_rev_constraints = newsize;
+        return 0;
+    } else {
+        free(db->rev_constraints);
+        return 1;
+    }
+}
+
+void init_reverse_constraints(struct dbtable *db)
+{
+    db->rev_constraints = calloc(INITREVCONSTRAINTS, sizeof(constraint_t *));
+    db->cap_rev_constraints = INITREVCONSTRAINTS;
+    db->n_rev_constraints = 0;
+}
+
+int add_reverse_constraint(struct dbtable *db, constraint_t *cnstrt)
+{
+    int rc = 0;
+    if (db->n_rev_constraints >= db->cap_rev_constraints) {
+        if ((rc = resize_reverse_constraints(db, 2 * db->cap_rev_constraints)) != 0) {
+            return rc;
+        };
+    }
+    db->rev_constraints[db->n_rev_constraints++] = cnstrt;
+    return rc;
+}
+
+int delete_reverse_constraint(struct dbtable *db, size_t idx, constraint_t *cnstrt)
+{
+    int rc = 0;
+    if (idx > db->n_rev_constraints) {
+        return 1;
+    }
+    if (cnstrt) {
+        *cnstrt = *db->rev_constraints[idx];
+    }
+    db->rev_constraints[idx] = NULL;
+
+    for (size_t i = idx; i < db->n_rev_constraints - 1; ++i) {
+        db->rev_constraints[i] = db->rev_constraints[i + 1];
+        db->rev_constraints[i + 1] = NULL;
+    }
+
+    db->n_rev_constraints--;
+
+    if (db->n_rev_constraints > 0 && db->n_rev_constraints <= db->cap_rev_constraints / 4) {
```

Comment:
resizing is not super-important.

Reply:
Since, this is tested and working, I don't want to remove this retest. So, leaving this here for now.